docs/reference/python/v0/storage/bucket-file-signurl.md

```diff
@@ -0,0 +1,72 @@
+---
+title: bucket.file.signUrl()
+description: Create a signed URL for access to a file.
+---
+
+Create a signed url for access to a file.
+
+```python
+from nitric.resources import bucket
+
+assets = bucket('assets').allow('reading', 'writing')
+
+logo = assets.file('images/logo.png')
+
+download_url = await logo.download_url()
+upload_url = await logo.upload_url()
+```
+
+## Parameters
+
+---
+
+**expiry** optional `number`
+Seconds until link expiry. Defaults to `600`, Maximum of `604800` (7 days)
+
+---
+
+## Examples
+
+### Create a readable link that is valid for the next 5 minutes
+
+```python
+from nitric.resources import bucket
+
+assets = bucket('assets').allow('reading')
+
+logo = assets.file('images/logo.png')
+
+logo_url = await logo.download_url(expiry=300)
+```
+
+### Create a temporary file upload link for a user
+
+```python
+from nitric.resources import bucket
+
+uploads = bucket('uploads').allow('writing')
+
+photo = assets.file('images/photo.png')
+
+photo_url = await logo.upload_url()
+```
+
+### Get an image url for rendering
+
+```python
+from nitric.resources import api, bucket
+from nitric.faas import HttpContext
+from nitric.application import Nitric
+
+main_api = api('main')
+images = bucket('images').allow('reading')
+
+@main_api.get('/images/:id')
+async def get_image(ctx: HttpContext):
+  id = ctx.req.params['id']
+  url = await images.file(id).download_url()
+  ctx.res.status = 303
+  ctx.res.headers['Location'] = [url]
```

Comment:
Should this also include an expires header with a comment on why its good to include it?

Reply:
thats a good suggestion, have spoken about the same thing to Rak, might include in a separate guide or best practcies guide, but it probably couldn't hurt to have it in these examples as well. The main issue is how verbose the timestamp formatting code will be for the header.